playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/OtherNavComponent.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { VisualGuidelinesItems } from "../MenuData/GuidelinesNavItems";
+import { GuidesNavItems } from "../MenuData/GuidesNavItems";
+import RoutedNavItem from "../RoutedNavItem";
+
+export const OtherNavItems = ({
+  name,
+  dark,
+  samples,
+  setIsActive,
+  updateTopLevelNav,
+  parentIndex,
+  reloadDocument
+}) => {
+  //transform text from samples yml
+  const transformMenuTitle = (link) => {
+    if (name === "UI Samples") {
+      const words = link
+        .split("_")
+        .map((word) => word.charAt(0).toUpperCase() + word.slice(1));
+      return words.join(" ");
+    } else {
+      return link.name;
+    }
+  };
+
+  const samplesMenu: string[] = [];
+
+  //Get samples pages from Samples yml file
+  for (const key in samples) {
+    if (samples.hasOwnProperty(key)) {
+      samples[key].forEach((item: string) => samplesMenu.push(item));
+    }
+  }
+  let menuItems: { [key: string]: string }[] | string[] = [];
+
+  //conditionally render navitems depending on name
+  if (name === "Tokens & Guidelines") {
+    menuItems = VisualGuidelinesItems;
+  } else if (name === "UI Samples" && samples) {
+    menuItems = samplesMenu;
+  } else if (name === "Getting Started") {
+    menuItems = GuidesNavItems;
+  }
```

Comment:
Consider using an object map here instead of conditionals:

```tsx
objMap = {
  "Tokens & Guidelines": VisualGuidelinesItems,
  "UI Samples": samplesMenu,
  "Getting Started": GuidesNavItems
}

menuItems = objMap[name]
```

Doing this would also potentiate the ability to make use of dependency injection.

Reply:
100%. Good call.